commonItems/SystemUtils.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace commonItems {
+    public static class SystemUtils {
+        public static SortedSet<string> GetAllFilesInFolder(string path) {
+            if (DoesFolderExist(path)) {
+                return new SortedSet<string>(Directory.GetFiles(path));
+            }
+            return new SortedSet<string>();
+        }
+        public static SortedSet<string> GetAllFilesInFolderRecursive(string path) {
+            if (DoesFolderExist(path)) {
+                return new SortedSet<string>(Directory.GetFiles(path, "*.*", SearchOption.AllDirectories));
+            }
+            return new SortedSet<string>();
+        }
+
+        public static SortedSet<string> GetAllSubfolders(string path) {
+            if (DoesFolderExist(path)) {
+                return new SortedSet<string>(Directory.GetDirectories(path));
+            }
+            return new SortedSet<string>();
+        }
+
+        public static bool TryCreateFolder(string path) {
+            try {
+                return Directory.CreateDirectory(path).Exists;
+            } catch (Exception e) {
+                Logger.Log(LogLevel.Error, "Could not create directory: " + path + " : " + e);
+                return false;
```

Comment:
Isn't it more idiomatic in C# to rethrow the exception instead of converting to a success/fail return type?

Reply:
Same for the other functions, possibly even functions in past PRs

Reply:
Returning success is also used, so I will use it for now to stay close to C++ version. Rethrowing in these functions would be useful if we wanted to have advanced handling in the converter for specific exceptions, but does any converter do that?
I've changed the names to `TryX` for consistency, though.